libs/amount/amount.cpp

```diff
@@ -100,15 +90,26 @@ namespace amount {
     return *this;
   }
 
+  // to raise to power integer values
+  int Amount::ipow(int base, int exp) const {
```

Comment:
what's the point of using your own impl? use `boost::multiprecision::pow` or `std::pow` instead

Reply:
std::pow returns float or double. I only need to raise to positive powers, so I need only integers to be returned. Floats are not exact and multiplying huge amounts may cause problems. That is why I've tried to avoid using floats here.

Reply:
Then use boost's one